routing/blinding.go

```diff
@@ -124,15 +133,33 @@ func NewBlindedPaymentPathSet(paths []*BlindedPayment) (*BlindedPaymentPathSet,
 	// replacement, so our target pub key in this case just remains the
 	// real introduction node ID.
 	for _, path := range paths {
-		if len(path.BlindedPath.BlindedHops) != 1 {
-			continue
-		}
+		pathLength := len(path.BlindedPath.BlindedHops)
 
-		pathSet = []*BlindedPayment{path}
-		finalCLTVDelta = path.CltvExpiryDelta
-		targetPub = path.BlindedPath.IntroductionPoint
+		if pathLength == 1 {
+			pathSet = []*BlindedPayment{path}
+			finalCLTVDelta = path.CltvExpiryDelta
+			targetPub = path.BlindedPath.IntroductionPoint
+
+			break
+		}
 
-		break
+		// We add a dummy hop to the end of the path so that
+		// we can do MPP path finding taking all the blinded
+		// routes into account. Moreover we need to use a
+		// dummy hop here because we still want to be able to
+		// penalize the last hop in case the payment fails.
+		// We need to set the cipher text because it is used for the
+		// payload size estimation.
+		lastHopCipherText := path.BlindedPath.BlindedHops[pathLength-1].
+			CipherText
+
+		path.BlindedPath.BlindedHops = append(
```

Comment:
This might be a bit critical, because we are altering the path which is add via a pointer, so probably we should do a deep copy here ?

Reply:
Decided to copy the `BlindedHops` before appending an element.

Reply:
hmmm im not sure this `copy(...)` does a deep  copy...? think it just copies over the pointers afaik

Reply:
actually - i dont understand why editing the path is a problem since we are doing it anyways?

Reply:
so the general idea of this approach is, to not alter the input paramter here, because normally you do not expect that because there is a return value to this function signature and changing the input value is not good design imo. Moreover allow to alter the existing blinded_hop slice, would also not allow us to run some unit tests in parallel, because we are reusing global defined blindedpaths.

Reply:
no but i mean you make a good point re not altering the passed in param. I think that's a good idea - we should just do the full copy then